src/main/java/org/project36/qualopt/web/rest/StudyResource.java

```diff
@@ -157,20 +159,76 @@ else if (authentication.getPrincipal() instanceof String) {
     }
 
     /**
-     * POST  /studies/send : send the study.
+     * POST  /studies/sendToAll : send the study to all participants.
      *
      * @param study the study to send
      * @return the ResponseEntity with status 201 (Created) if the study was sent or 400 (Bad Request) if the study doesn't exist
      */
-    @PostMapping(path = "/studies/send",
+    @PostMapping(path = "/studies/sendToAll",
         produces={MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE})
     @Timed
-    public ResponseEntity sendStudy(@Valid @RequestBody Study study) {
-        log.debug("REST request to send Study : {}", study);
+    public ResponseEntity sendStudyToAll(@Valid @RequestBody Study study) {
+    	
+    	log.debug("REST request to send Study : {}", study);
         if (Objects.isNull(study)){
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-        studyService.sendInvitationEmail(study);
+        
+        /* "true" mean "ToAll = True", send to all participants
+         * No need for working with a copy,
+         * Unlike "sendStudyToNew", "sendInvitationEmail" won't look at 
+         * "EmailAddressesHaveInvited" list, so a update on that won't effect 
+         * the result of "sendInvitationEmail"
+         */
+        studyService.sendInvitationEmail(study, true);	
+        
+        /*
+         * Next 4 lines: Mark this list of participants as "Have recieved email",
+         * by adding their email address into the list of EmailAddressesHaveInvited
+         */
+        for (Participant singleParticipant : study.getParticipants()){
+        	study.addToEmailAddressesHaveInvited(singleParticipant.getEmail());
+        }
+        studyRepository.save(study);
+        
+        return new ResponseEntity<>(HttpStatus.CREATED);
+    }
+    
+    /**
+     * POST  /studies/sendToNew : send the study to participants that haven't received emails
+     *
+     * @param study the study to send
+     * @return the ResponseEntity with status 201 (Created) if the study was sent or 400 (Bad Request) if the study doesn't exist
+     */
+    @PostMapping(path = "/studies/sendToNew",
+        produces={MediaType.APPLICATION_JSON_VALUE, MediaType.TEXT_PLAIN_VALUE})
+    @Timed
+    public ResponseEntity sendStudyToNew(@Valid @RequestBody Study study) {
+    	
+    	log.debug("REST request to send Study : {}", study);
```

Comment:
Should this be here?

Reply:
what do you mean "This"?, the whole function?